src/DirectoryCleaner.php

```diff
@@ -4,22 +4,23 @@
 
 use Carbon\Carbon;
 use Illuminate\Support\Collection;
-use Illuminate\Filesystem\Filesystem;
+use Illuminate\Support\Facades\Storage;
 
 class DirectoryCleaner
 {
-    /** @var \Illuminate\Filesystem\Filesystem */
+    /** @var \Illuminate\Contracts\Filesystem\Filesystem */
     protected $filesystem;
 
     /** @var string */
     protected $directory;
 
     /**
-     * @param \Illuminate\Filesystem\Filesystem $filesystem
+     * DirectoryCleaner constructor.
      */
-    public function __construct(Filesystem $filesystem)
+    public function __construct()
```

Comment:
Please keep on using dependency injection here. Use a service provider to inject the right filesystem driver.